api/v1beta2/authpolicy_types.go

```diff
@@ -141,6 +141,18 @@ type AuthPolicySpec struct {
 	// +kubebuilder:validation:XValidation:rule="self.kind == 'HTTPRoute' || self.kind == 'Gateway'",message="Invalid targetRef.kind. The only supported values are 'HTTPRoute' and 'Gateway'"
 	TargetRef gatewayapiv1alpha2.PolicyTargetReference `json:"targetRef"`
 
+	// Defaults define explicit default values for this policy and for policies inheriting this policy.
+	// Defaults are mutually exclusive with implicit defaults defined by CommonSpec.
+	// +optional
+	Defaults *CommonSpec `json:"defaults"`
+
+	// CommonSpec defines implicit default values for this policy and for policies inheriting this policy.
+	// CommonSpec is mutually exclusive with explicit defaults defined by Defaults.
+	CommonSpec `json:""`
+}
+
+// CommonSpec contains common shared fields for defaults and overrides
+type CommonSpec struct {
```

Comment:
This gives issues with CEL due to and size of the CRD

```
The CustomResourceDefinition "authpolicies.kuadrant.io" is invalid: 
* metadata.annotations: Too long: must have at most 262144 bytes
* spec.validation.openAPIV3Schema.properties[spec].properties[rules].properties[authentication].additionalProperties.properties[routeSelectors].items.properties[matches].items.properties[path].x-kubernetes-validations[10].rule: Forbidden: contributed to estimated rule cost total exceeding cost limit for entire OpenAPIv3 schema
* spec.validation.openAPIV3Schema.properties[spec].properties[rules].properties[authorization].additionalProperties.properties[routeSelectors].items.properties[matches].items.properties[path].x-kubernetes-validations[10].rule: Forbidden: contributed to estimated rule cost total exceeding cost limit for entire OpenAPIv3 schema
* spec.validation.openAPIV3Schema.properties[spec].properties[rules].properties[callbacks].additionalProperties.properties[routeSelectors].items.properties[matches].items.properties[path].x-kubernetes-validations[10].rule: Forbidden: contributed to estimated rule cost total exceeding cost limit for entire OpenAPIv3 schema
* spec.validation.openAPIV3Schema.properties[spec].properties[rules].properties[metadata].additionalProperties.properties[routeSelectors].items.properties[matches].items.properties[path].x-kubernetes-validations[10].rule: Forbidden: contributed to estimated rule cost total exceeding cost limit for entire OpenAPIv3 schema
* spec.validation.openAPIV3Schema: Forbidden: x-kubernetes-validations estimated rule cost total for entire OpenAPIv3 schema exceeds budget by factor of 1.7x (try simplifying the rule, or adding maxItems, maxProperties, and maxLength where arrays, maps, and strings are declared)
make: *** [install] Error 1
```

So we might want to rethink of whether we truely want the 2 pathways to achieving defaults 🤔

Reply:
Added a workaround at https://github.com/Kuadrant/kuadrant-operator/pull/503/commits/6a01a4b870343c293c70159822517adf96819043 decreasing the max number of route selectors allowed, but would likely hit this problem again for overrides also which would decrease this number again